agsol-common/src/max_len_string.rs

```diff
@@ -14,6 +14,12 @@ impl<const N: usize> MaxSerializedLen for MaxLenString<N> {
     const MAX_SERIALIZED_LEN: usize = 4 + N;
 }
 
+impl<const N: usize> MaxLenString<N> {
```

Comment:
I'd say a
`pub fn contents(&self) -> &str {
    self.contents.as_str()
}`
would be a better option to get read access to the contents. And if needed you can always call `self.contents().to_owned()` to get an owned version. Furthermore, we could add a
`pub fn contents_mut(&mut self) -> &mut str {
    self.contents.as_mut_str()
}`
that allows in-place modifications to the string, without worrying about modifying the length of the underlying contents.